docs/DeveloperGuide.md

```diff
@@ -2,29 +2,310 @@
+
+Empty of unrecognized command:
+When the returned string is empty, we simply continue to the next iteration of the loop. This is essentially similar to a "pass" statement
+as when the user does not input anything, the application silently waits for the user to input something into the command line. 
+When there is an unrecognized command, the application prompts the user through a function call to a static method printInvalidMainMenuCommandMessage()
+(belongs to AllOnUs class and not included in the diagrams above) to enter a known command and suggests referring to the help section.
+
+Exit command:
+If the isExitCommand() method returns true on the string representing user input, the loop breaks, and control is returned to the static main()
+method of the AllOnUs class, which then calls the static exit method in the same class to print a termination message, and then finally control
+is returned to the OS. 
+
+### Study Manager component
+API: `StudyManager.java`
+
+![](images/ModuleClassDiagram.png)
+
+
+The `StudyManager` component,
+1. Stores the academic schedule, i.e. all `Module` objects are contained in a `modulesList` object.
+2. The storage can be bound to `modulesList` list such that everytime a change is observed in the list it is saved on to a file.
+3. Does not depend on other components like `ExpenseTracker` and `ContactsManager`
+
+How the StudyManager works:
+1. Based on a command entered by the user, StudyManager can perform the following functions 
+   1. add 
+   2. list
+   3. delete
+   4. edit
+   5. find 
+   6. read ics
+2. There are two ways to add to the `moduleList` to keep track of academic schedule 
+   1. Either the user can choose to add the modules one by one. (See user guide for add feature)
+   2. Or the user can choose to read from NUSMods .ics calendar file of their academic schedule.  (See user guide for read from ics feature)
+      1. This is more convenient as it can add multiple modules at a single go.
+      2. This feature makes use of the `ModuleCalendarReader.java` API.
+3. If there are any errors in the entry of `Module` parameters they are handled by the various exceptions in the `exceptions` package.
+
+   
+The two ways to add modules are illustrated in the sequence diagram below:
+
+
+![](images/StudyManagerSequenceDiagram.png)
+
+
+### Expense Component
+
+![](images/ExpenseClassDiagram.png)
+
+How the Expense Tracker component works:
+1. From the Main Menu, if the user decides to run `goto m/Expense_Tracker`, the `ExpenseTracker` class takes over.
+2. The `expenseRunner` method begins by taking in user inputs depending on the user's requirements.
+3. Depending on the user's input, the `ExpenseParser` class calls 
+`parseXYZExpense` where `XYZ` is a placeholder for the specific command name (eg. `parseDeleteExpense`).
+4. After parsing, the command is executed accordingly within the `ExpenseTracker` class.
+
+#### [Proposed] Budgeting Reminder Feature
```

Comment:
perhaps can make the header a bit bigger?
![image](https://user-images.githubusercontent.com/75602488/161185914-a2605b16-2bc2-4729-9d6f-70d923f9c735.png)